### PR TITLE
Feature/create user player instance

### PR DESCRIPTION
**Pull request by ArnaudPoletto**

### CODE REVIEW

File: app/src/androidTest/java/sdp/moneyrun/LoginInstrumentedTest.java [hunk 1]
```diff
@@ -167,29 +283,64 @@ public void loginWithRegisteredUserStartsActivity(){
...
+            assertNull(FirebaseAuth.getInstance().getCurrentUser());
+
+            db.removePlayer(playerUser);
             Intents.release();
         } catch (InterruptedException e) {
-            e.printStackTrace();
-            Intents.release();
-        }
+        e.printStackTrace();
+        Intents.init();
```
Comment (Alvaro2112) [inline app/src/androidTest/java/sdp/moneyrun/LoginInstrumentedTest.java:343]: Shouldnt this be Intents.release()?

File: app/src/main/java/sdp/moneyrun/DatabaseProxy.java [hunk 2]
```diff
@@ -29,9 +29,25 @@ public DatabaseProxy(){
      * @param player the player to be put in the database
      */
     public void putPlayer(Player player){
+        if(player == null){
+            throw new IllegalArgumentException("player should not be null");
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/DatabaseProxy.java:33]: Basic test, should be tested.

File: app/src/main/java/sdp/moneyrun/DatabaseProxy.java [hunk 3]
```diff
@@ -29,9 +29,25 @@ public DatabaseProxy(){
...
+
         mDataBase.child("players").child(String.valueOf(player.getPlayerId())).setValue(player);
     }
 
+    /**
+     * Remove a player to the database.
+     * @param player the player to be removed in the database
+     */
+    public void removePlayer(Player player){
+        if(player == null){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/DatabaseProxy.java:44]: Same here

File: app/src/main/java/sdp/moneyrun/MenuActivity.java [hunk 4]
```diff
@@ -219,17 +215,27 @@ private void setNavigationViewListener() {
...
         startActivity(playerProfileIntent);
     }
+
+    public void onButtonSwitchToLeaderboardActivity() {
+        Intent leaderboardActivity = new Intent(MenuActivity.this, LeaderboardActivity.class);
+        setPutExtraArguments(leaderboardActivity);
+        startActivity(leaderboardActivity);
+    }
+
+    public void onButtonSwitchToLoginActivity(){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/MenuActivity.java:233]: I think when going back to the login activity (Logging out), the current activity should be closed (ie. finish())

File: app/src/main/java/sdp/moneyrun/PlayerProfileActivity.java [hunk 5]
```diff
@@ -26,33 +26,29 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            }
-        });
+
         Intent playerIntent = getIntent();
-        int playerId = playerIntent.getIntExtra("playerId",0);
-        String[] playerInfo = playerIntent.getStringArrayExtra("playerId"+playerId);
-        setDisplayedTexts(playerInfo);
+        Player user = (Player) playerIntent.getSerializableExtra("user");
+
+        goBackToMain.setOnClickListener(v -> {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/PlayerProfileActivity.java:33]: Should be tested

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/sdp/moneyrun/LoginInstrumentedTest.java (hunk 1 above)
Comment (ArnaudPoletto) [reply app/src/androidTest/java/sdp/moneyrun/LoginInstrumentedTest.java]: yes, fixed

File: app/src/main/java/sdp/moneyrun/DatabaseProxy.java (hunk 2 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/DatabaseProxy.java:33]: done

File: app/src/main/java/sdp/moneyrun/DatabaseProxy.java (hunk 3 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/DatabaseProxy.java:44]: done

File: app/src/main/java/sdp/moneyrun/MenuActivity.java (hunk 4 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/MenuActivity.java]: indeed, done

File: app/src/main/java/sdp/moneyrun/PlayerProfileActivity.java (hunk 5 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/PlayerProfileActivity.java:33]: done

**Review by Alvaro2112 — APPROVED** [review]

**Review by gauthierboeshertz — APPROVED** [review]

**Pull request merged**